vcf.c

```diff
@@ -145,12 +145,34 @@ int bcf_hdr_add_sample(bcf_hdr_t *h, const char *s)
 
 int HTS_RESULT_USED bcf_hdr_parse_sample_line(bcf_hdr_t *h, const char *str)
 {
+    static int hdr_space_warned = 0;
     int ret = 0;
     int i = 0;
     const char *p, *q;
     // add samples
     for (p = q = str;; ++q) {
-        if ((unsigned char) *q > '\n') continue;
+        if ((unsigned char) *q == ' ')
+        {
+            if ( i==0 && strncmp(p,"#CHROM",q-p) ) hts_log_error("Could not parse the sample line: %s",str);
+            if ( i==1 && strncmp(p,"POS",q-p) ) hts_log_error("Could not parse the sample line: %s",str);
+            if ( i==2 && strncmp(p,"ID",q-p) ) hts_log_error("Could not parse the sample line: %s",str);
+            if ( i==3 && strncmp(p,"REF",q-p) ) hts_log_error("Could not parse the sample line: %s",str);
+            if ( i==4 && strncmp(p,"ALT",q-p) ) hts_log_error("Could not parse the sample line: %s",str);
+            if ( i==5 && strncmp(p,"QUAL",q-p) ) hts_log_error("Could not parse the sample line: %s",str);
+            if ( i==6 && strncmp(p,"FILTER",q-p) ) hts_log_error("Could not parse the sample line: %s",str);
+            if ( i==7 && strncmp(p,"INFO",q-p) ) hts_log_error("Could not parse the sample line: %s",str);
+            if ( i==8 && strncmp(p,"FORMAT",q-p) ) hts_log_error("Could not parse the sample line: %s",str);
+            if ( !hdr_space_warned ) hts_log_warning("The sample names contain spaces, is this intended? Trying to proceed anyway..");
```

Comment:
I assume you added ".." so it the function can add the final ".", but it triggers a failure the test harness.  Maybe just omit the ".." to silence this.

Reply:
The dots ".." are removed now